src/main/java/org/lagonette/hellos/service/PurgeDatabaseService.java

```diff
@@ -30,7 +30,7 @@ public void purgeDatabase() {
         LOGGER.info("Begin of the database purge");
 
         // Payment purge
-        List<Integer> paymentsToDelete = paymentRepository.findIdsByInsertionDateBefore(LocalDateTime.now().minusDays(nbOfDaysForPayments));
+        List<Integer> paymentsToDelete = paymentRepository.findIdsByInsertionDateBefore(LocalDateTime.now().minusDays(dotenv.get("DATABASE_RETENTION_PAYMENT_DAY")));
```

Comment:
should cast to long, and check if constante is a long instance

Reply:
Pbm de compilation dessus d'après Maven 
```
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project hellos: Compilation failure
Error:  /home/runner/work/hellos/hellos/src/main/java/org/lagonette/hellos/service/PurgeDatabaseService.java:[33,129] incompatible types: java.lang.String cannot be converted to long
```

Reply:
Done